code/src/uart_nmea.cpp

```diff
@@ -0,0 +1,104 @@
+#include <cmath>
+#include <cstdint>
+#include <string.hpp>
+#include <uart_nmea.hpp>
+
+namespace r2d2::location {
+    uart_nmea_c::uart_nmea_c(r2d2::usart::usart_connection_c &usart_port)
+        : listener(usart_port) {
+    }
+
+    frame_coordinate_s uart_nmea_c::coordinates_to_frame(float longitude,
+                                                         float latitude,
+                                                         bool north, bool east,
+                                                         int16_t altitude) {
+
+        frame_coordinate_s frame;
+
+        frame.altitude = altitude;
+
+        frame.long_deg = static_cast<uint8_t>(longitude / 100);
```

Comment:
The only thing these coordinates are used for is for sending on the bus and they get converted to an integer here. Is there any point in storing them as floats in the first place? Removing all these floating point operators would vastly improve performance.

Reply:
In my point of view there was no need to store anything in floats. The previous team made this decision. To remove this we might need to ask permission to the PO first. But this can be removed yes.

Reply:
We got permission from PO to change the floats to something else. For now I removed most of the calculation's with floats. For this PR I will leave the rest of the floats. This might change in the future but for now we don't have the time to change the rest.